### ISSUE TITLE
about the MoE

### ISSUE THREAD

**Issue by Ideal-Bushgorman** [YuqiYang213/MLoRE#6]
Hi ! It seems that every task has a set of experts, since  self.conv2[task] = LoraBlock(final_embed_dim, final_embed_dim, kernel_size=kernel_size, rank=p.spe_rank).

`class MOEBlock(nn.Module):
    def __init__(self, p, final_embed_dim, im_size, kernel_size=1, with_feat=False):
        super().__init__()
        self.num_lora = len(p.rank_list)
        self.p = p
        self.lora_list_1 = nn.ModuleList()
        rank_list = p.rank_list
        for i in range(self.num_lora):
            self.lora_list_1.append(LoraBlock(final_embed_dim, final_embed_dim, kernel_size=kernel_size, rank=rank_list[i]))
            self.lora_list_1[i].init_weights()
        self.conv1 = nn.ModuleDict()
        self.conv2 = nn.ModuleDict()
        self.conv3 = nn.ModuleDict()
        self.share_conv = nn.Conv2d(final_embed_dim, final_embed_dim, kernel_size=3, padding=1)
        self.bn = nn.ModuleDict()
        self.bn_all = nn.ModuleDict()
        self.activate = nn.GELU()
        for task in self.p.TASKS.NAMES:
            self.conv1[task] = nn.Conv2d(final_embed_dim, final_embed_dim, kernel_size=1)
            self.conv3[task] = nn.Conv2d(final_embed_dim, final_embed_dim, kernel_size=1)
            self.conv2[task] = LoraBlock(final_embed_dim, final_embed_dim, kernel_size=kernel_size, rank=p.spe_rank)`

**Comment by YuqiYang213** [comment]
Sorry for the late reply. 
The `self.conv2` in our module is the **Task-specific low-rank expert path** mentioned in Sec 3.3 in our paper, which can enhance the distinctiveness of the task-specific features.
The expert networks in MoE are the `self.lora_list_1`.

**Comment by Ideal-Bushgorman** [comment]
> Sorry for the late reply. The `self.conv2` in our module is the **Task-specific low-rank expert path** mentioned in Sec 3.3 in our paper, which can enhance the distinctiveness of the task-specific features. The expert networks in MoE are the `self.lora_list_1`.

Thank you
